File: src/com/venus/Queue/Queue.java
package com.venus.Queue;

import com.venus.List.DoublyLinkedList;

/**
 * @copyright 熊彪
 *  * 队列
 *  * 注意点：可以使用ArrayList或者DoublyLinkedList来实现，为什么？
 *          ArrayList与DoublyLinkedList在头尾节点操作的时间复杂度均为O（1）
 *  *
 */
@SuppressWarnings("all")
public class Queue<E> {

    private DoublyLinkedList<E> list = new DoublyLinkedList<>();

    public void enqueue(E element){
        list.add(element);
    }

    public E dequeue() {
        return list.remove(0);
    }

    public E front() {
        return list.get(0);
    }

    public void clear() {
        list.clear();
    }

    public boolean isEmpty() {
        return list.isEmpty();
    }

    public int size() {
        return list.size();
    }
}

File: Python/Queue/queue.py
## written by xiongbiao
## date 2020-5-19

'''
队列思路
1. 线性结构
2. 顺序相同
'''
class Queue(object):
    def __init__(self):
        self.__queue = []

    def clear(self):
        self.__queue = []

    def size(self):
        return len(self.__queue)

    def is_empty(self):
        return self.size() == 0

    def enqueue(self, value):
        self.__queue.append(value)

    def dequeue(self):
        if self.size() == 0:
            raise IndexError('queue is empty, can not pop value')
        return self.__queue.pop(0)

    def front(self):
        return self.__queue[0]

    def __repr__(self):
        s = 'Queue: ['
        for i, n in enumerate(self.__queue):
            if i == 0:
                s += '{}'.format(n)
            else:
                s += ', {}'.format(n)
        s += ']'
        return s
#
# queue = Queue()
# queue.enqueue(1)
# queue.enqueue(2)
# queue.enqueue(3)
# queue.enqueue(5)
# queue.dequeue()
# queue.dequeue()
# queue.dequeue()
# queue.dequeue()
# queue.dequeue()
# print(queue)